src/DAO/GenericDAO.php

```
@@ -224,20 +224,21 @@ private function bindArrayValue(\PDOStatement $query, $array, $typeArray = false
     /**
      * {@inheritDoc}
      */
-    public function generate(IDataBaseCreator $creator, $create = false)
+    public function geraScriptDeCriacaoDoBancoDeDados(IDatabaseCreator $creator)
     {
         $script = $creator->scriptCreation($this->type, true);
         
-        if ($create == false) {
-            return $script;
-        } else {
-            // TODO extract to method
-            try {
-                $dbh = new PDO(DB_DSN, DB_USERNAME, DB_PASSWORD, array( PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
-                $dbh->exec($script);// or die(print_r($dbh->errorInfo(), true));
-            } catch (PDOException $e) {
-                echo ("DB ERROR: ". $e->getMessage());
-            }
+        return $script;
+    }
+
+    public function criaBancoDeDados(IDatabaseCreator $creator)
+    {
+        $script = $this->geraScriptDeCriacaoDoBancoDeDados($creator);
+        try {
```

Remove the `try`, let it fail instead.